packages/ui-app/src/Params/Param/Balance.tsx

```diff
@@ -0,0 +1,192 @@
+// Copyright 2017-2018 @polkadot/ui-app authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { ApiProps } from '@polkadot/ui-react-rx/types';
+import { I18nProps } from '../../types';
+import { Props as BareProps } from '../types';
+
+import BN from 'bn.js';
+import React from 'react';
+import withApi from '@polkadot/ui-react-rx/with/api';
+
+import isValidBalance from '../../util/isValidBalance';
+import Bare from './Bare';
+import InputNumber from '../../InputNumber';
+import translate from '../../translate';
+
+type Props = I18nProps & ApiProps & BareProps;
+
+type State = {
+  error?: React.ReactNode,
+  info?: React.ReactNode
+};
+
+class Balance extends React.PureComponent<Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props);
+
+    this.state = {
+      error: '',
+      info: ''
+    };
+  }
+
+  render () {
+    const { className, defaultValue: { value }, isError, label, style, t, withLabel } = this.props;
+    const { error, info } = this.state;
+    const defaultValue = new BN((value as BN).toString(10) || '0').toString(10);
+
+    const maxLengthForLatestChainSpec = 39;
+
+    return (
+      <Bare
+        className={className}
+        style={style}
+      >
+        <InputNumber
+          className='large'
+          defaultValue={defaultValue || '0'}
+          error={error}
+          info={info}
+          isError={isError}
+          label={label}
+          maxLength={maxLengthForLatestChainSpec}
+          onChange={this.onChange}
+          onKeyDown={this.onKeyDown}
+          onKeyUp={this.onKeyUp}
+          placeholder={t('account.balance.placeholder', {
+            defaultValue: 'Between 1 testnet DOT and the available testnet DOT balance (minus 1) of the account'
+          })}
+          withLabel={withLabel}
+        />
+      </Bare>
+    );
+  }
+
+  onChange = (value: string): void => {
+    const { onChange, t } = this.props;
+
+    // prepare value by remove all whitespace
+    value = value.split(' ').join('');
+
+    try {
+      const { isValid, errorMessage, infoMessage, inputConvertedFromScientificNotation } = isValidBalance(value);
+
+      this.setState({
+        error: !isValid && errorMessage ? t(errorMessage) : '',
```

Comment:
???

The way this is done -

`t(<name-for-the-string>, { defaultValue: <default-english-value> })`

That is the only way it will make sense to translate since `name-for-the-string` gives content, where/what and `default-english-value` shows the translator what the original was.

Reply:
EDIT: Technically the above would work, however where the concern comes in - the mapping now matches against the full text, instead of a static key that should not change. If we now make any changes to that string (which is used in translation matching), the already translated text breaks. (So something as simple as a type breaks anything that has been translated for that text.)

Always stick to key + defaultValue everywhere.